src/libutil/util.cc

```diff
@@ -232,7 +232,11 @@ DirEntries readDirectory(const Path & path)
         checkInterrupt();
         string name = dirent->d_name;
         if (name == "." || name == "..") continue;
+#ifdef HAVE_STRUCT_DIRENT_D_TYPE
         entries.emplace_back(name, dirent->d_ino, dirent->d_type);
+#else
+        entries.emplace_back(name, dirent->d_ino, getFileType(absPath(name, path)));
```

Comment:
I think this can be:
```
entries.emplace_back(name, dirent->d_ino, DT_UNKNOWN);
```
since callers of `readDirectory` have to handle the possibility of `DT_UNKNOWN` anyway, and we don't want to do a stat call for every directory entry unless it's really needed.